sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/TakeDocumentQueryExecutionContext.java

```diff
@@ -17,56 +18,92 @@
 import java.util.function.Function;
 import java.util.function.Predicate;
 
-public class TopDocumentQueryExecutionContext<T>
+public class TakeDocumentQueryExecutionContext<T>
     implements IDocumentQueryExecutionComponent<T> {
 
+    private static final Logger logger = org.slf4j.LoggerFactory.getLogger(TakeDocumentQueryExecutionContext.class);
     private final IDocumentQueryExecutionComponent<T> component;
-    private final int top;
-    // limit from rewritten query
+    private final int takeCount;
+    private final TakeEnum takeEnum;
 
-    public TopDocumentQueryExecutionContext(IDocumentQueryExecutionComponent<T> component, int top) {
+    public TakeDocumentQueryExecutionContext(IDocumentQueryExecutionComponent<T> component, int take, TakeEnum takeEnum) {
         this.component = component;
-        this.top = top;
+        this.takeCount = take;
+        this.takeEnum = takeEnum;
     }
 
-    public static <T> Flux<IDocumentQueryExecutionComponent<T>> createAsync(
-            BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<IDocumentQueryExecutionComponent<T>>> createSourceComponentFunction,
-            int topCount,
-            int limit,
-            String topContinuationToken,
-            PipelinedDocumentQueryParams<T> documentQueryParams) {
-        TakeContinuationToken takeContinuationToken;
+    public static <T> Flux<IDocumentQueryExecutionComponent<T>> createTopAsync(BiFunction<String, PipelinedDocumentQueryParams<T>, Flux<IDocumentQueryExecutionComponent<T>>> createSourceComponentFunction,
```

Comment:
`createAsync` static method is a common pattern followed by all the QueryExecutionContext classes. Maybe you could have `createAsync` and switch between top or limit inside it?